### ISSUE TITLE
Complete lack of documentation.

### ISSUE THREAD

**Issue by marcinant** [sigma/gh.el#36]
Is there any chance to get some usage examples?

The only hint I can see is available in 'gh-issues.el'.

Unfortunately it's wrong.
1. there is no 'gh-issues-list' but only an 'gh-issues-issue-list'
2. I corrected this error above but the only thing I get as result is: 'byte-code: JSON readtable error'

How should I configure this library? Where should I provide and store github password and token?

I can run all tests properly. So this library is workable. But is there any real life use case?

**Comment by aspiers** [comment]
In principal having OO elisp is great, but since starting to play with `gh.el` in the last few hours, my experience has been that eieio makes debugging a bit of a nightmare.  Maybe I'm just dumb when it comes to understanding macrostep or something.

**Comment by sigma** [comment]
@aspiers you're absolutely right, it's a nightmare (although to be fair, gh.el itself is partly responsible, and I'm trying to remove a lot of code in the "next" branch). Especially since eieio is not particularly great at maintaining compatibility from version to version right now (so things tend to fail in very obscure ways).
I guess in the long run I still hope that the debuggers will learn enough about eieio to become more useful.

**Comment by jakoblind** [comment]
One year since last comment here, and the issue is still relevant. I try to figure out how to use this library but the docs are very limited. Where can I enter the oauth credentials? I am confused. Is this project still alive?

**Comment by sigma** [comment]
@jakoblind well, I would definitely welcome patches to improve the documentation, but I'm not even sure that would address your question as it would be the "developer" kind of documentation most likely (gh.el is not meant to be directly user-facing).
Anyway, `gh-auth.el` (and `gh-oauth.el`) is where you'd see that the oauth token is stored in the git configuration property `github.oauth-token`. But that should be taken care of automatically...

**Comment by jakoblind** [comment]
I came here trying to figure out how to make `magit-gh-pulls` work with personal access tokens (https://github.com/settings/tokens). The code is very difficult for me to get into, I might just write what I need myself :) thanks for your help anyway!

**Comment by wpcarro** [comment]
After some searching, I found [this link](http://ghel.readthedocs.io/en/latest/).

Would this be worth linking to in the Readme?
